tests/api.rs

```diff
@@ -388,8 +388,169 @@ mod tests {
+              # General
+              type = "blackhole"
+              inputs = ["events_processed_total_batch_source"]
+              print_amount = 100000
+        "#,
+        )
+        .await;
+
+        let server = api::Server::start(topology.config());
+        let client = new_subscription_client(server.addr()).await;
+        let subscription = client.component_events_processed_total_batch_subscription(500);
+
+        tokio::pin! {
+            let component_events_processed_total_batch = subscription.stream();
+        }
+
+        let data = component_events_processed_total_batch
+            .next()
+            .await
+            .unwrap()
+            .unwrap()
+            .data
+            .unwrap()
+            .component_events_processed_total_batch;
+
+        assert_eq!(data[0].name, "events_processed_total_batch_source");
+        assert_eq!(data[1].name, "events_processed_total_batch_sink");
+
+        assert_eq!(
+            data[0].metric.events_processed_total,
+            data[1].metric.events_processed_total
+        );
+    }
+
+    #[tokio::test]
+    #[allow(clippy::float_cmp)]
+    #[ignore]
+    /// Tests componentBytesProcessedTotal returns increasing metrics, ordered by
+    /// source -> transform -> sink
+    async fn api_graphql_component_bytes_processed_total() {
+        init_metrics();
+
+        let topology = from_str_config(
+            r#"
+            [api]
+              enabled = true
+
+            [sources.bytes_processed_total_source]
+              type = "generator"
+              lines = ["Random line", "And another"]
+              batch_interval = 0.1
+
+            [sinks.bytes_processed_total_sink]
+              # General
+              type = "blackhole"
+              inputs = ["bytes_processed_total_source"]
+              print_amount = 100000
+        "#,
+        )
+        .await;
+
+        let server = api::Server::start(topology.config());
+        let client = new_subscription_client(server.addr()).await;
+        let subscription = client.component_bytes_processed_total_subscription(500);
+
+        tokio::pin! {
+            let component_bytes_processed_total = subscription.stream();
+        }
+
+        // Results should be sorted by source -> sink, so we'll need to assert that
+        // order. The events generated should be the same in both cases
+        let mut map = HashMap::new();
+
+        for r in 0..=1 {
+            map.insert(
+                r,
+                component_bytes_processed_total
+                    .next()
+                    .await
+                    .unwrap()
+                    .unwrap()
+                    .data
+                    .unwrap()
+                    .component_bytes_processed_total,
+            );
+        }
+
+        // Should only show byte totals on sinks
+        assert_eq!(map[&0].name, "bytes_processed_total_sink");
+        assert_eq!(map[&0].name, "bytes_processed_total_sink");
```

Comment:
nit:  duplicate assertion here

Reply:
Thanks! That second one should pulling from `map[&1].name`.